server/src/hover.rs

```diff
@@ -39,6 +40,18 @@ impl HoverVisitor {
         }
     }
 
+    fn construct_message<T: Element + ?Sized>(
```

Comment:
Doesn't seem that this can ever return `None` We can remove the `Option` return.